django_freeradius/api/views.py

```diff
@@ -16,20 +20,40 @@
 User = get_user_model()
 
 
-@api_view(['POST'])
-def authorize(request):
-    username = request.data.get('username')
-    password = request.data.get('password')
-    try:
-        user = User.objects.get(username=username, is_active=True)
-    except User.DoesNotExist:
-        user = None
-    if user and user.check_password(password):
-        return Response({'control:Auth-Type': 'Accept'}, status=200)
-    return Response({'control:Auth-Type': 'Reject'}, status=401)
+class TokenAuthentication(BaseAuthentication):
+    def authenticate(self, request):
+        if request.META.get('HTTP_AUTHORIZATION', False):
+            headers = request.META.get('HTTP_AUTHORIZATION').split(',')
+            for header in headers:
+                token = header.split(' ')[1]
+                if token == API_TOKEN:
+                    return (AnonymousUser(), None)
+        if request.GET.get('token', False):
+            if request.GET.get('token') == API_TOKEN:
```

Comment:
can we simplify these two nested `if` by doing simply:

```python
if request.GET.get('token') == API_TOKEN:
    # ...
```

If the token is not set the logic will never get here because we are raising `ImproperlyConfigured`.